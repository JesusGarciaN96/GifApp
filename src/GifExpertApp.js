import React, { useState } from 'react';

import AddCategory from './components/AddCategory/AddCategory';
import GifGrid from './components/GifGrid/GifGrid';

function GifExpertApp() {
  const [categories, setCategories] = useState([]);

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory addCategory={setCategories} />
      <hr />
      {categories.map(cate => (
        <GifGrid key={cate} category={cate} />
      ))}
    </>
  );
}

export default GifExpertApp;
