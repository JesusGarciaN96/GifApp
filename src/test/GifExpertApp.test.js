import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifExpertApp from '../GifExpertApp';

describe('Test de la vista de gifs', () => {
  test('El render de la vista debe mostrarse', () => {
    const componente = shallow(<GifExpertApp />);
    expect(componente).toMatchSnapshot();
  });
});
