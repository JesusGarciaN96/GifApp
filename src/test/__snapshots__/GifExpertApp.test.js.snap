// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de la vista de gifs El render de la vista debe mostrarse 1`] = `
<Fragment>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Giphy Application
  </h1>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
</Fragment>
`;
